core/src/main/scala/org/apache/spark/HttpFileServer.scala

```diff
@@ -20,6 +20,7 @@ package org.apache.spark
 import java.io.File
 
 import com.google.common.io.Files
+import org.apache.commons.httpclient.util.URIUtil
```

Comment:
Before you consider doing more work to change again, I'd love to get another opinion if possible. We won't stall too long on this in any event. I suppose I preferred your original change, yes, to this one, just because it avoids any new subtle dependency entanglement.